magicl.asd

```diff
@@ -137,9 +137,12 @@
    :type "f"))
 
 (defmethod output-files ((operation compile-op) (component f->so))
-  (values (list (make-pathname :name "libexpokit"
-                               :type #-darwin "so" #+darwin "dylib"
-                               :defaults (component-pathname component)))
+  (values (list (apply-output-translations
+                 (make-pathname :name "libexpokit"
+                                :type #+darwin "dylib"
```

Comment:
How about:

```
(defun dynamic-library-extension ()
  (cond
    ((uiop:os-windows-p) "dll")
    ((uiop:os-macosx-p)  "dylib")
    ((uiop:os-unix-p)    "so")
    (t                   (error "unsupported OS"))))
```

then call that function here without any of these conditionals?

(sorry, I probably should have offered this in my first round of review)